specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1178 @@
+
+// Retrieve the activation args using Windows App SDK AppLifecycle functions
+AppActivationArguments args { AppLifecycle::AppInstance::GetCurrent().GetActivatedEventArgs() };
+ExtendedActivationKind kind { args.Kind() };
+
+// Check the activation kind
+if (kind == ExtendedActivationKind::AppNotification)
+{
+    AppNotificationActivatedEventArgs appActivatedArgs { args.Data().as<AppNotificationActivatedEventArgs>() };
+
+    // Retrieve the serialized arguments from AppNotificationActivatedEventArgs
+    winrt::Windows::Foundation::Collections::IMap<winrt::hstring, winrt::hstring> argumentMap { appActivatedArgs.Argument() };
+}
+```
+
+# Full API Details
+
+```cpp
+namespace Microsoft.Windows.AppNotifications.Builder
+{
+    runtimeclass AppNotificationTextProperties
+    {
+        // Contains the set of <text> attributes
+        AppNotificationTextProperties();
+
+        AppNotificationTextProperties SetLanguage(String value);
+        AppNotificationTextProperties UseIncomingCallAlignment();
+        AppNotificationTextProperties SetMaxLines(Int32 value);
+
+        String GetXml();
+    }
+
+    enum AppNotificationButtonStyle
+    {
+        Success,
+        Critical,
+    };
+
+    runtimeclass AppNotificationButton
+    {
+        // Button can use ToolTip instead of content which requires empty content.
+        // Argument attribute is required and uses the ArgumentSerializer construct
+        // to add arguments to the button.
+        AppNotificationButton();
+        AppNotificationButton(String buttonText);
+
+        AppNotificationButton AddArgument(String key, String value);
+
+        // Sets the IconUri for the button.
+        AppNotificationButton SetIcon(Windows.Foundation.Uri value);
+
+        // The tooltip for a button, if the button has an empty content string.
+        AppNotificationButton SetToolTip(String value);
+
+        // Sets the Button as context menu action.
+        AppNotificationButton UseContextMenuPlacement();
+
+        // Sets the AppNotificationButtonStyle to Success or Critical
+        AppNotificationButton SetButtonStyle(AppNotificationButtonStyle value);
+
+        // Sets the ID of an existing TextBox in order to have this button display to the right of the input.
+        AppNotificationButton SetInputId(String value);
+
+        // Launches the URI passed into the button when activated.
+        AppNotificationButton SetInvokeUri(Windows.Foundation.Uri protocolUri);
+        AppNotificationButton SetInvokeUri(Windows.Foundation.Uri protocolUri, String targetAppId);
+
+        // Retrieves the XML content of the button.
+        String GetXml();
+    };
+
+    runtimeclass AppNotificationComboBox
+    {
+        AppNotificationComboBox(String id);
+
+        // Add a selection item to the AppNotificationComboBox if InputType is AppNotificationComboBox. The id parameter is used to retrieve the user input when the app is activated
```

Comment:
The `id` doesn't so much "retrieve the user input" as it "identifies the item the user selected".